src/mount.azfs

```diff
@@ -266,172 +258,25 @@ update_ipmap()
-                echo "[BUG] Blob endpoint ip ($l_nfsip) is invalid!" 
-                continue 
-            fi 
-
- 
-            # 
-            # Local share name. Note that we mount the local IP and not the actual Blob endpoint IP and 
-            # map local IP to actual Blob endpoint IP using DNAT rule. 
-            # 
-
-            m_share="$l_ip:$l_dir" 
-          
-            # 
-            # Delete entry from IPMAP if share is unmounted. 
-            # TODO: What if user unmounts and mounts before MONITOR_INTERVAL_SECS secs? 
-            # 
-
-            if ! echo "$findmnt" | grep " nfs $m_share " > /dev/null; then 
-                # delete the line from IPMAP file. 
-                echo "[$date] Deleting unmounted share $m_share" 
-                chattr -i $IPMAP 
-                sed -i "/ $l_ip $l_nfsip/d" $IPMAP 
-                chattr +i $IPMAP 
-                iptables -t nat -D OUTPUT -p tcp -d "$l_ip" -j DNAT --to-destination "$l_nfsip" 
-            fi 
- 
-            # 
-            # Check if blob endpoint IP address changed. 
-            # This is the migration check. 
-            # 
-
-            new_ip=$(resolve_ipv4 "$l_host") 
- 
-            # If we fail to resolve the host name, try next time. 
-
-            if [ $? -ne 0 ]; then 
-                echo "resolve_ipv4($l_host) failed!" 
-                continue 
-            fi 
- 
-            if ! is_valid_ipv4_address "$new_ip"; then 
-                echo "resolve_ipv4($l_host) returned invalid IPv4 address ($new_ip)!" 
-                continue 
-            fi 
- 
-            # 
-            # If the IP changed for the Blob endpoint, we need to update the DNAT rule. 
-            # This will take care of migration/failover causing the Blob endpoint IP to change. 
-            # 
-            # TODO: Make this less frequent than the unmount check. 
-            # 
-            #echo "$l_host: new_ip=$new_ip, l_nfsip=$l_nfsip" 
-
-            if [ "$new_ip" != "$l_nfsip" ]; then 
-                echo "IP for $l_host changed [$l_nfsip -> $new_ip]" 
-                chattr -i $IPMAP 
-                sed -i "/ $l_ip $l_nfsip/d" $IPMAP 
-                chattr +i $IPMAP 
- 
-                # Delete old DNAT rule and the conntrack entry to stop current active connections too. 
-                iptables -t nat -D OUTPUT -p tcp -d "$l_ip" -j DNAT --to-destination "$l_nfsip" 
-                conntrack -D conntrack -p tcp -d "$l_ip" -r "$l_nfsip" 
- 
-                # Add new DNAT rule. 
-
-                iptables -t nat -A OUTPUT -p tcp -d "$l_ip" -j DNAT --to-destination "$new_ip" 
-                chattr -i $IPMAP 
-                echo "$l_host:$l_dir $l_ip $new_ip" >> $IPMAP 
-                chattr +i $IPMAP 
-            fi 
-        done 
-    done 
-} 
+}
 
- 
 
 ##### Action starts here ##### 
- 
+
 # [52.239.178.215:/nfsv3stdaccount20/fstestbase3 /mnt/azfs -o rw,tcp,nolock,nconnect=16] 
 echo "Got arguments: [$*]" 
- 
+
 # 
```

Comment:
now we don't need this code that deletes the IPMAP file and tries to kill old watchdog.
It should just check that the watchdog service is running (using "systemctl status azfswatchdog") and if not bail out with an appropriate error, but not try to kill.

Reply:
removed.